### PR TITLE
[Merged by Bors] - chore: Make sure `WithOne` doesn't import rings

### PR DESCRIPTION
**Pull request by YaelDillies**
Reorder the `WithOne` material.
* `Algebra.Group.WithOne.Defs` was hiding a `Ring` import! I credit Johan and Mario for https://github.com/leanprover-community/mathlib/pull/2707.
* `WithBot` is not needed to define `WithOne.unone`. It's simpler to redefine it by hand. In the future, we might want to have an `Option` version (but not sure how much that's worth, since it would basically be `Option.get` again).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/WithOne/Defs.lean [hunk 1]
```diff
@@ -119,8 +123,7 @@ attribute [elab_as_elim] WithZero.recZeroCoe
 /-- Deconstruct an `x : WithOne α` to the underlying value in `α`, given a proof that `x ≠ 1`. -/
 @[to_additive unzero
       "Deconstruct an `x : WithZero α` to the underlying value in `α`, given a proof that `x ≠ 0`."]
-def unone {x : WithOne α} (hx : x ≠ 1) : α :=
-  WithBot.unbot x hx
+def unone : ∀ {x : WithOne α}, x ≠ 1 → α | (x : α), _ => x
```
Comment (eric-wieser) [inline Mathlib/Algebra/Group/WithOne/Defs.lean:126]: What's the motivation for this change?

File: Mathlib/Algebra/Group/WithOne/Defs.lean [hunk 2]
```diff
@@ -375,34 +377,12 @@ instance addMonoidWithOne [AddMonoidWithOne α] : AddMonoidWithOne (WithZero α)
...
-    WithZero.monoidWithZero, WithZero.instDistrib with }
-
 end WithZero
 
 -- Check that we haven't needed to import all the basic lemmas about groups,
 -- by asserting a random sample don't exist here:
 assert_not_exists inv_involutive
 assert_not_exists div_right_inj
 assert_not_exists pow_ite
+
```
Comment (sgouezel) [inline Mathlib/Algebra/Group/WithOne/Defs.lean:387]:
```suggestion
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Algebra/Group/WithOne/Defs.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Group/WithOne/Defs.lean:387]: This is on purpose: The comment above only applies to the three previous `assert_not_exists`, not to this one.

**Comment by YaelDillies** [comment]
bors merge
